File: client/src/utilities/types/Transaction.ts
import { ISpace } from ".";

type ITransactionTypeRequest = {
  value: string;
};

type ITransactionType = {
  id: string;
  created_datetime: string;
  updated_datetime: string;
  value: string;
};

type ITransactionRequest = {
  type: ITransactionType['id'];
  amount: number;
  occurred_datetime: Date;
};

type ITransaction = {
  id: string;
  created_datetime: string;
  updated_datetime: string;
  occurred_datetime: string;
  type: ITransactionType;
  amount: number;
  space: ISpace;
};

export default ITransaction;
export type { ITransactionType, ITransactionRequest, ITransactionTypeRequest };

File: server/schemas/Transaction.py
from uuid import UUID
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel, ConfigDict

from schemas.Space import SpaceResponse


class TransactionTypeRequest(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    value: str

class TransactionTypeResponse(TransactionTypeRequest):
    model_config = ConfigDict(from_attributes=True)

    id: UUID


class TransactionRequest(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    type: UUID
    amount: Decimal
    occurred_datetime: datetime


class TransactionResponse(TransactionRequest):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    space: SpaceResponse
    type: TransactionTypeResponse
    created_datetime: datetime
    updated_datetime: datetime
